datafusion/core/src/datasource/listing/helpers.rs

```diff
@@ -153,225 +151,239 @@ pub fn split_files(
         .collect()
 }
 
+struct Partition {
```

Comment:
I think adding some comments about specifically what `depth` and `files` fields mean would help readability. Like are the files only files or do they include paths and what does `depth` signify?